ddl/backfilling.go

```diff
@@ -1067,12 +1072,18 @@ func (dc *ddlCtx) writePhysicalTableRecord(sessPool *sessionPool, t table.Physic
 	}
 
 	for {
+		if startKey.Cmp(endKey) >= 0 {
```

Comment:
Could we remove this check?

Reply:
No, `kvRanges[len(kvRanges)-1].EndKey` may equal to `endKey`.

The function `splitTableRanges()` doesn't handle the cases like `start_key <= end_key`.

Reply:
Got it.

Reply:
is above  start_key > end_key scenario for `splitTableRanges()` not handle?

Reply:
It handles in [line1082](https://github.com/pingcap/tidb/pull/41460/files#diff-1bbc7833321782b3a22a842b6237974b6a60cb07b49b246b85552ab5e915ed6eR1082)

Reply:
My question is if we send a smaller start_key to `splitTableRanges(), will it return an err?